edkrepo/common/common_repo_functions.py

```diff
@@ -832,11 +833,25 @@ def create_local_branch(name, patchset, global_manifest_path, manifest_obj, repo
                 repo.remotes.origin.fetch(patchset.fetch_branch, progress=GitProgressHandler())
             except:
                 raise EdkrepoFetchBranchNotFoundException(FETCH_BRANCH_DOES_NOT_EXIST.format(patchset.fetch_branch))
+            parent_patchsets = list()
             try:
-                parent_patchset = manifest_obj.get_patchset(patchset.parent_sha, patchset.remote)
-                repo.git.checkout(parent_patchset[2], b=name)
-            except:
-                if patchset.parent_sha in repo.tags:
+                parent_patchsets.append(manifest_obj.get_patchset(patchset.parent_sha, patchset.remote))
+                while True:
+                    # Allow daisy-chaining PatchSet.  Continue until KeyError is caught.
+                    parent_patchsets.append(manifest_obj.get_patchset(parent_patchsets[len(parent_patchsets)-1].parent_sha, parent_patchsets[len(parent_patchsets)-1].remote))
+                    if parent_patchsets.count(parent_patchsets[len(parent_patchsets)-1]) > 1:
+                        # Do not continue to branch creation step if a circular dependency is detected between multiple PatchSet.
+                        raise ValueError(PATCHSET_CIRCULAR_DEPENDENCY_ERROR.format(parent_patchsets[len(parent_patchsets)-1]))
+            except KeyError:
+                if parent_patchsets != list():
+                    base_patchset_parent_sha = parent_patchsets[len(parent_patchsets)-1].parent_sha
+                    if base_patchset_parent_sha in repo.tags:
+                        # tag names are prioritized over branch names
+                        repo.git.checkout("tags/{}".format(base_patchset_parent_sha), b=name)

```

Comment:
Thinking about this, we should probably use the fully qualified path of "refs/tags/{}"... just in case some joker decides it is a good idea to make a branch called "tags/ourtagname".

I realize that you are just doing what the existing code does, but it is always best to leave things better than we found them.

Reply:
I tested with your suggested change, and I don't see any regression in behavior.

The only new issue I can find is if I make a branch with the same name as a commit SHA, then use that commit SHA as the PatchSet parentSha... then the branch is used instead of the commit SHA for the Dynamic Branch Creation.

git branch aea8a9c954a79029ded443eadf7c92716c1753cc ff8485179c1cb847bff17725fd95043976108a54

    <PatchSet remote="Edk2Repo" name="edk2_test_branch" parentSha="aea8a9c954a79029ded443eadf7c92716c1753cc" fetchBranch="master">
    </PatchSet>